packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -36,18 +37,86 @@ enum _ScaffoldSlot {
   statusBar,
 }
 
+/// Geometry information for scaffold components.
+///
+/// To get a [ValueNotifier] for the scaffold geometry call
+/// [Scaffold.geometryOf].
+@immutable
+class ScaffoldGeometry {
+  const ScaffoldGeometry({
+    this.bottomNavigationBarTop,
+    this.floatingActionButtonPosition,
+    this.floatingActionButtonScale = 1.0,
+  });
+
+  /// The distance from the scaffold's top edge to the top edge of the
+  /// rectangle in which the [Scaffold.bottomNavigationBar] bar is being laid
+  /// out.
+  ///
+  /// When there is no [Scaffold.bottomNavigationBar] set, this will be null.
+  final double bottomNavigationBarTop;
+
+  /// The rectangle in which the scaffold is laying out
+  /// [Scaffold.floatingActionButton].
+  ///
+  /// The floating action button might be scaled inside this rectangle, to get
+  /// the bounding rectangle in which the floating action is painted scale this
+  /// value by [floatingActionButtonScale].
```

Comment:
It's not clear what "scale" means here. Maybe show same `## Sample code` for how to do that?

Reply:
done.
@Hixie looking at it again I don't like the Position suffix we added to floatingActionButton, as it sounds like a point and not a rectangle. What do you think about just calling it floatingActionButton (geometry.floatingActionButton sounds pretty reasonable I believe).
If not, maybe floatingActionButtonRect ?

Reply:
A "floatingActionButton" property should hold a floating action button.
I agree that "position" suggests an Offset.

This gives you the area that the FAB would cover if it was not scaled, right?
Maybe "floatingActionButtonArea"?